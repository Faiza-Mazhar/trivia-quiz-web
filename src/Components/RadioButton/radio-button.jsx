import React from "react";
import "./radio-button.style.scss";
const RadioButton = ({ name, value, onClick }) => {
  return (
    <label className="card">
      {value}
      <input
        type="radio"
        name={name}
        value={value}
        id={value}
        onClick={onClick}
      />
    </label>
  );
};

export default RadioButton;
