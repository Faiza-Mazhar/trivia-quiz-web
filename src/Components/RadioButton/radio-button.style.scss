.radio-button-label {
  padding: 10px;

  display: flex;
  cursor: pointer;
  background-color: #e2eee2;
  color: #2a3121;
  font-weight: bold;
  font-size: large;
  justify-content: space-between;
  border: 1px solid #e2eee2;
  border-radius: 5px;
}

.radio {
  display: none;
}

input[type="radio"]:after {
  width: 15px;
  height: 15px;
  border-radius: 15px;
  top: -2px;
  left: -2px;
  position: relative;
  background-color: #d1d3d1;
  content: "";
  display: inline-block;
  visibility: visible;
  border: 2px solid #abb899;
}

input[type="radio"]:checked:after {
  width: 15px;
  height: 15px;
  border-radius: 15px;
  top: -2px;
  left: -2px;
  position: relative;
  background-color: #556342;
  content: "";
  display: inline-block;
  visibility: visible;
  border: 2px solid #abb899;
}
