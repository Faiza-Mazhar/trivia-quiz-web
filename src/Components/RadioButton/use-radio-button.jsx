import { useState } from "react";

const useRadioButtons = (name, handleChange) => {
  const [value, setValue] = useState(null);

  const handleClick = ({ target }) => {
    setValue(target.value);
    handleChange(target.value);
  };

  const inputProps = {
    name,
    type: "radio",
    onClick: handleClick,
  };

  return [value, inputProps];
};

export default useRadioButtons;
