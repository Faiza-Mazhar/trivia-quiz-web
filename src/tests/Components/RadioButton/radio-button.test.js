import RadioButton from "../../../Components/RadioButton/radio-button";

import renderer from "react-test-renderer";

test("Radio buttons, if no props are set", () => {
  const component = renderer.create(<RadioButton />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Radio buttons, if with props", () => {
  const component = renderer.create(
    <RadioButton
      value={"value"}
      inputProps={{
        name: "name",
        type: "radio",
        onClick: jest.fn,
      }}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
