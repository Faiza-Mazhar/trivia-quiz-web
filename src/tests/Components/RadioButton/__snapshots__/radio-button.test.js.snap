// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio buttons, if no props are set 1`] = `
<label
  className="radio-button-label"
>
  <input />
</label>
`;

exports[`Radio buttons, if with props 1`] = `
<label
  className="radio-button-label"
>
  value
  <input
    name="name"
    onClick={[Function]}
    type="radio"
    value="value"
  />
</label>
`;
